src/ipc/dbus/DBusObjectRegistry.cpp

```diff
@@ -49,15 +64,28 @@ void DBusObjectRegistry::init()
     if (!m_initialized) {
         m_initialized = true;
         if (m_dbusManager.registerServiceName(m_serviceName)) {
-            m_master = std::make_unique<MasterImpl>(*this);
+            m_master = std::make_unique<MasterImpl>();
             m_master->init();
-            QObject::connect(m_master.get(), &MasterImpl::objectsChanged, this, &DBusObjectRegistry::objectsChanged);
+            QObject::connect(m_master.get(), &MasterImpl::objectAdded, this, &DBusObjectRegistry::onObjectAdded);
+            QObject::connect(m_master.get(), &MasterImpl::objectRemoved, this, &DBusObjectRegistry::onObjectRemoved);
         } else {
-            m_objectRegistryAsyncProxy = new facelift::ipc::dbus::ObjectRegistryAsyncIPCDBusProxy();
+            m_objectRegistryAsyncProxy = new ObjectRegistryAsyncIPCDBusProxy(this);
             m_objectRegistryAsyncProxy->ipc()->setServiceName(m_serviceName);
             m_objectRegistryAsyncProxy->connectToServer();
-            QObject::connect(m_objectRegistryAsyncProxy, &facelift::ipc::dbus::ObjectRegistryAsync::objectsChanged, this,
-                    &DBusObjectRegistry::objectsChanged);
+
+            QObject::connect(m_objectRegistryAsyncProxy, &ObjectRegistryAsync::readyChanged,
+                             m_objectRegistryAsyncProxy, [this]() {
+                                 m_objectRegistryAsyncProxy->getObjects(facelift::AsyncAnswer<QMap<QString, QString>>(
+                                     this, [this](QMap<QString, QString> objectMap) {
+                                         if (m_objectRegistryAsyncProxy->ready()) {
```

Comment:
why testing readiness here ?

Reply:
This was meant to be before getObjects. Fixing.